src/Couchbase.Lite.Shared/Sync/WebSocketWrapper.cs

```diff
@@ -638,10 +637,17 @@ private void SetupAuth()
         {
             var auth = _options?.Auth;
             if (auth != null) {
-                var username = auth.Username;
-                var password = auth.Password;
-                if (username != null && password != null) {
-                    _logic.Credential = new NetworkCredential(username, password);
+                if (auth.Type == AuthType.HttpBasic) {
+                    var username = auth.Username;
+
+                    // TODO string Password will be deprecated and replaced with byte array password
+                    var password = auth.PasswordData != null ?
+                         Encoding.Unicode.GetString(auth.PasswordData)
+                         : auth.Password;
```

Comment:
Use the `SecureString` constructor of `NetworkCredential` for the non-string password